core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -7810,9 +7810,6 @@ public void _testGroupExpressionEquivalenceParams() {
     sql("select * from emp where deptno = ?").ok();
     sql("select * from emp where deptno = ? and sal < 100000").ok();
     sql("select case when deptno = ? then 1 else 2 end from emp").ok();
-    // It is not possible to infer type of ?, because SUBSTRING is overloaded
-    sql("select deptno from emp group by substring(ename from ^?^ for ?)")
-        .fails("Illegal use of dynamic parameter");
```

Comment:
@julianhyde maybe I am missing something but why was it impossible to infer type of `?`. The overload is only for the first argument but not the second and the third according to https://calcite.apache.org/docs/reference.html#character-string-operators-and-functions

Reply:
It's a bit strange to comment on your own PR. Let's talk in the JIRA case.

Reply:
Updated the PR. PTAL. Thanks